### ISSUE TITLE
Adjust tray icon behaviour on XFCE.

### ISSUE THREAD

**Issue by tio-trom** [borgbase/vorta#1206]
This might sound stupid, but I am using it on Arch via the AUR package and if I click the close button it simply minimizes it ti the tray icon area. There I have no right click options. So how do I close Vorta?

**Comment by m3nu** [comment]
You can fully quit it from the system tray menu. There should also be an option und *Misc* that decides if it will be fully closed when closing the window.

I should note that some desktop envs don't properly show the system tray menu.  If that's the case for yours, we can look into making some adjustments.

**Comment by tio-trom** [comment]
I am using XFCE + Arch. Weirdly enough Vorta is the only app so far that shows no menu if I interact with it from the tray panel, and I have tested thousands of apps so far (I curate linux apps too for a project of mine).

In the Misc tab I do not see any option about closing the window:
![2022-02-20_20-00](https://user-images.githubusercontent.com/38043315/154859526-01a86e3e-ff49-4f6a-9b36-113c6609eb5d.png)

**Comment by m3nu** [comment]
"Display background exit dialog" But not sure this is what you need.

Planning to change the tray behavior for XFCE. See here for full discussion: #938

**Comment by real-yfprojects** [comment]
> Weirdly enough Vorta is the only app so far that shows no menu if I interact with it from the tray panel

Have you used the right mouse button?

**Comment by tio-trom** [comment]
Yes. No mouse button works....

**Comment by real-yfprojects** [comment]
On my VM, the right mouse button will open a context menu provided by xfce. Is that the case for you too?

There is also an option in the preferences of the *status notifier plugin* called 1Menu is primary action`. Disabling the option makes the intended behaviour of vorta's tray icon working. However, one still cannot open vorta's context menu by right clicking. 
If the option is enabled, the plugin claims that a left click will open the menu but it seems like this isn't working with Qt.

I therefore suggest that vorta shows the context menu when being activated on xfce.

**Comment by real-yfprojects** [comment]
> "Display background exit dialog" But not sure this is what you need.

I think this one is broken because a don't see any behavioural changes when toggling the option.

**Comment by m3nu** [comment]
Can do. The desktop env name is in an env var we already use.

**Comment by tio-trom** [comment]
In my XFCE none works. I need to disable the "!Menu is primary action" to be able to open it again from the tray else I can't even bring the window to focus....

**Comment by phil294** [comment]
> In my XFCE none works. I need to disable the "!Menu is primary action" to be able to open it again from the tray else I can't even bring the window to focus....

This is the same behavior for me (xfce, aur). And regardless of all above mentioned options, closing the app only works via task manager or pkill. For context, here's what right click menu looks like:

![img](https://waritschlager.de/share/973286AA452C5858861F58306C83535B.png)

with "Properties" only pointing to general status tray plugin settings.

All other "status tray plugin" entries I tested shipped their very own menu, like for example optimus-manager-qt:

![img](https://waritschlager.de/share/DB1EC51CD7313FEA9AA25FC4E26933EB.png)

And this is what most native (non-status tray plugin) xfce items look like, e.g. pulseaudio:

![img](https://waritschlager.de/share/A9C15DD8A497D860DDD0691E00405533.png)

Hope this helps

**Comment by a544jh** [comment]
Can confirm I have the same issue with Vorta 0.8.7 from Debian bullseye-backports with XFCE.

**Comment by alelapp** [comment]
Same issue with Vorta 0.8.9 from Xubuntu 22.04.1 (either installed via apt and pip3). Vorta process has to be killed by terminal.

**Comment by m3nu** [comment]
I don't use Xfce myself, but you can open a PR any time. The change will probably go somewhere here:

https://github.com/borgbase/vorta/blob/master/src/vorta/tray_menu.py#L25

**Comment by real-yfprojects** [comment]
> It is a small inconvenience but quite frustrating to not be able to close it somehow. I opened this issue some 1 year ago, I hope the Vorta devs can add a fix for this sooner. :D

Easy changes like this one are tagged `good first issue` to signal that they suit new contributors. In our very limited time we core developers try to fix more difficult issues.

**Comment by tio-trom** [comment]
I understand guys. The reason I and others hope for a solution soonish is because this is a big issue in XFCE that makes you unable to close Vorta unless you know how to kill the process via a different program. I wish I could help. I do a lot of work on other free projects too, but I can't have the skills to help with this one.

**Comment by m3nu** [comment]
The work here isn't making the change, it's setting up a desktop with XFCE.

I did remember that I run some virtual desktops using XFCE for another project, so might as well use those. Can you summarize what you need again and could you test the changes to verify? Meaning to install a feature branch via pip, as mentioned here: https://vorta.borgbase.com/contributing/#testing-the-current-development-version

**Comment by RubenKelevra** [comment]
This is just limited to XFCE, as under Gnome the tray icon got a context menu (with extension to activate tray icons).
